### DISCUSSION TITLE
Changes to the Brawler skill tree

### DISCUSSION THREAD

**Discussion by NoIWontBruh0831** [FakeFishGames/Barotrauma#11854] [Ideas and suggestions]
### Brawler's Competition
The Brawler class's talents have a skewed range from being extremely good to pretty terrible. An issue contributing to this is that classes like the Engineer, Security Officer, and Assistant all have much better bonuses to melee builds. 
- Engineer has triple damage Crowbars, the PUCS, and the risky yet powerful Volatile Fulgurium Fuel Rods. 
- Security Officer has Swole which is a great all-around melee buff, Stonewall which is an extremely powerful talent that is a better version of one of Brawler's, and Inordinate Exsanguination/ which both give strong bonuses for either survivability or damage. 
- Assistant has Psycho Clown and Graduation Ceremony. Both of these are better than a majority of Mechanic's melee talents. Psycho Clown grants 2.5x attack speed for being under Psychosis. This, in my opinion, is an amazing risk-reward talent, unlike some of Brawler's talents. Graduation Ceremony unlocks the single best melee weapon in the entire game. Engineers and Mechanics both cannot benefit from the weapon as much, but Security Officers can. Engineers not benefitting as much isn't too much of a problem for them as their triple damage Crowbars are incredibly powerful anyway.

---

### Brawler's Talents
**I Am That Guy**
- This is probably the best talent for this skill tree. It gives an amazing weapon/tool for Mechanics along with nice buffs. Multifunctional is required to make this usable as a weapon though, as it only deals 30 damage per hit without it. An issue with the Heavy Wrench is that the quality bonuses only affect its repair speed, not damage. Because this item is basically a Mechanic's main melee weapon by design, not having a damage boost from quality bonuses makes this weapon and, by proxy, the entire Brawler class less viable. This weapon also has extremely poor damage against armor.

**Heavy Lifting**
- This talent is alright at best. A small 20% movement speed increase that only procs when carrying Metal Crates and Artifacts isn't impactful at all. This isn't useful for you or your crew nor does it help a melee build.

**Mudraptor Wrestler**
- A decent talent IF you hit the weak points of a Mudraptor. Because the Heavy Wrench has such poor armor penetration, this talent increases damage dealt to a Mudraptor's armor by **1%**. This is an issue with the Heavy Wrench, not the talent itself, but still worth noting.

**Berserker and Foolhardy**
- I'm putting these talents together because their usages are identical. These talents are extremely terrible, Berserker being the much worse one. The issue with these talents is that they require you to be half-dead and grant negligible buffs. I love risk-reward talents but these 2 are just horrible. Foolhardy gives a 20% physical damage bonus when below 50% which doesn't even matter because you'd already be bleeding out. Berserker's 20% MELEE damage bonus when under 50% isn't even noticeable. The main issue is that these buffs are so tiny for the risk you are putting yourself at when reaching the 50% HP threshold.

**MECHanic**
- I like this talent to be honest. I know many people dislike Exosuits and prefer PUCS suits, but the added damage resistances and storage the Exosuit give are nice to me. Someone else pointed this out, but because the Exosuit is so slow (even with this talent) if you ever get into a losing fight when far from the sub, you cannot escape and are forced to stand your ground. This is fine for me because it makes underwater ventures more interesting and is a smart way to balance a powerful diving suit.

**Issues with this skill tree**
Brawler's talents do not benefit melee builds as much as other classes, despite being EXCLUSIVELY melee. Engineer's talents give arguably the best suit and fuel rods in the game. Security Officer does the melee build better than Mechanic along with craftable guns the entire crew can use. The Assistant has other miscellaneous buffs along with a better melee build than Mechanic. Brawler's Exosuits are fine, but many people prefer PUCS and Slipsuit suits over them. 

---

### Changes to Brawler talents that could help it be more viable

**I Am That Guy**
- This talent is already great, but the Heavy Wrench needs quality bonuses that also affect damage. This item is basically designed to be a Mechanic's main melee weapon. It also needs better damage against armor.

**Heavy Lifting**
- Expand its effect to ALL large non-weapon items in the game (Ammo Boxes, Shells, Crates, etc) along with being able to hold them with one hand instead of two. Even with these changes. I feel like it wouldn't be chosen over Mudraptor Wrestler much. In my opinion, this talent doesn't make sense for the Brawler tree as it has 0 benefits to melee builds.

**Heavy Lifting rework**
- You now can grab ALL creatures in the game without the need for them to be stunned. While grabbing a creature, you gain a 30% stun resistance. You can drag dead and alive creatures up to the size of Hammerheads at full speed. This grab CANNOT be interrupted unless you receive heavy knockback or stun (Humans can interrupt your grab though). While grabbing the enemy, your cursor is the direction where you'll hold the enemy. Use this to hold enemies infront of you to act as a meatshield.
- Pressing E on a grabbed enemy throws the enemy to your cursor, stunning them for 1s and dealing additional damage with stun if they collide with an obstacle or another creature. Humans can also be thrown but will receive no damage. This throw's strength depends on how long you hold E. Holding E gives the affliction "Wind-up" and builds by 33% per second you hold E, up to 100% for double damage, throw speed, throw range, and stun.
- The damage and stun the thrown enemy receives when impacting a surface or creature depends on the velocity, and collided enemies and surfaces also receive damage. Heavier enemies such as Veteran Mudraptors and Hammerheads will not travel as far as Spinelings and Mudraptors when thrown, causing them to receive less damage and stun when impacting a surface or creature.

- Maybe this is too strong lol.

**Mudraptor Wrestler**
- To combat the Heavy Wrench's terrible armor pen, this talent now has a new ability. Every 5s, you gain the affliction "Suplex". 
- The Suplex affliction makes your next melee attack inflict 25% more stun to Mudraptors along with flipping them on their backs, exposing their weak flesh. This allows you to easily target and attack a Mudraptor's weak points every 5s.

**Berserker and Foolhardy**
- Berserker should grant WAY stronger buffs. I don't know what to suggest without making it way overtuned, but a 40% melee damage bonus and 25% melee attack speed bonus?
- Foolhardy should give much stronger defensive buffs. Again, I don't know what to suggest without making it overtuned, but maybe a 25% physical damage resistance along with a 40% stun resistance and a 20% movement speed increase? 

**MECHanic**
- I like this talent but many people dislike both this talent and the Exosuit. Maybe the Exosuit can let you fire ANY weapon without the necessary weapon skill? It could also have a brighter flashlight effect, not because it needs it or anything but because it would feel better to use.

These are changes I'd make for the Brawler skill tree. I probably didn't have to write so much lol

---

EDIT: Heavy Wrench should deal more damage against armored enemies.

- Against armored Mudraptors with the talents **I Am That Guy and Multifunctional**, you deal 3% Blunt Force Trauma towards their armor. With **Mudraptor Wrestler** added on, you deal 4% Blunt Force Trauma. This is so sad.

EDIT 2: New Mudraptor Wrestler effect lol
- Since the Heavy Wrench has a pretty poor armor pen, I thought making it easier to expose a Mudraptor's weak point through a wrestling move would be funny.

EDIT 3: Heavy Lifting rework
- Heavy Lifting's entire effect is useless for melee builds, so I added a silly rework that still makes use of the name "Heavy Lifting" which allows for powerful grabs.

EDIT 4: Gave the Heavy Lifting rework stun resistance
- Yeah.

**Comment by BiffW** [comment]
I think the heavy wrench is literally too small and needs to be changed bigger. And its attacking range needs to be increased.
I feel ok about its damage but a slight damage increase is welcome.

About Berserker and Foolhardy.
I have my ideas.

**Berserker**
For each 10% vitality loss, you gain 10% melee damage bonus, up to 50%

**Foolhardy**
For each 10% vitality loss, you gain 10% stun resistance, up to 50%

**Reply by BiffW** [reply]
About Foolhardy, I think mechanic don't need any more damage resistance since they can have exosuits later.
As a melee-focused talent branch, I think it's more important to not get stun-locked by swarm of enemies.

**Reply by NoIWontBruh0831** [reply]
This is better than what I had in mind lol. Thanks for also providing the code!

**Comment by BiffW** [comment]
Here are some xml documents of modified Berserker talent.
Text:
```
<talentdescription.onhealthpercentageloss>For each [healthloss]% vitality loss(up to [maxhealthloss]% loss), you gain:</talentdescription.onhealthpercentageloss>
```
Talent:
```

        <Talent identifier="berserker">
            <Icon texture="Content/UI/TalentsIcons3.png" sheetindex="3,6" sheetelementsize="128,128"/>
            <Description tag="talentdescription.onhealthpercentageloss">
                <Replace tag="[healthloss]" value="10" color="gui.green"/>
                <Replace tag="[maxhealthloss]" value="50" color="gui.green"/>
            </Description>
            <Description tag="talentdescription.additionalstattype">
                <Replace tag="[amount]" value="10" color="gui.green"/>
                <Replace tag="[stattype]" value="stattypenames.meleedamagebonus" color="gui.orange"/>
            </Description>
            <AbilityGroupEffect abilityeffecttype="OnAttack">
                <Conditions>
                    <AbilityConditionAttackData weapontype="melee" />
                    <AbilityConditionAboveVitality invert="true" vitalitypercentage="0.9"/>
                </Conditions>
                <Abilities>
                    <CharacterAbilityModifyAttackData addeddamagemultiplier="0.1" />
                </Abilities>
            </AbilityGroupEffect>
            <AbilityGroupEffect abilityeffecttype="OnAttack">
                <Conditions>
                    <AbilityConditionAttackData weapontype="melee" />
                    <AbilityConditionAboveVitality invert="true" vitalitypercentage="0.8"/>
                </Conditions>
                <Abilities>
                    <CharacterAbilityModifyAttackData addeddamagemultiplier="0.1" />
                </Abilities>
            </AbilityGroupEffect>
            <AbilityGroupEffect abilityeffecttype="OnAttack">
                <Conditions>
                    <AbilityConditionAttackData weapontype="melee" />
                    <AbilityConditionAboveVitality invert="true" vitalitypercentage="0.7"/>
                </Conditions>
                <Abilities>
                    <CharacterAbilityModifyAttackData addeddamagemultiplier="0.1" />
                </Abilities>
            </AbilityGroupEffect>
            <AbilityGroupEffect abilityeffecttype="OnAttack">
                <Conditions>
                    <AbilityConditionAttackData weapontype="melee" />
                    <AbilityConditionAboveVitality invert="true" vitalitypercentage="0.6"/>
                </Conditions>
                <Abilities>
                    <CharacterAbilityModifyAttackData addeddamagemultiplier="0.1" />
                </Abilities>
            </AbilityGroupEffect>
            <AbilityGroupEffect abilityeffecttype="OnAttack">
                <Conditions>
                    <AbilityConditionAttackData weapontype="melee" />
                    <AbilityConditionAboveVitality invert="true" vitalitypercentage="0.5"/>
                </Conditions>
                <Abilities>
                    <CharacterAbilityModifyAttackData addeddamagemultiplier="0.1" />
                </Abilities>
            </AbilityGroupEffect>
        </Talent>
```

**Comment by Joel-B-Williams** [comment]
I'd love to see Oxygenite also increase the speed of the exosuit (or maybe the quality of the fuel rod, if you want to keep the "this is a machine" vibe)
Alternatively, a boost mechanic to allow better propulsion that either uses a fire extinguisher or other item, or burns the fuel source more, etc

Mainly this is a QoL thing - moving so slow is drudgery.  I understand the trade-off between protection & speed, but moving this slow just makes basic tasks like mining & looting (where the storage is super helpful) a massive chore.  Any other suit + cargo scooter is far more suitable for the job

Part 2) a melee buff (mechanic-only) while wearing the suit would help the tree in general scale up, and make the suit more attractive compared to other options
Getting a buff with HMG's is interesting and makes some sense, but no other class relies on another class to get the most use out of their top-tier talents
